include/mav/Message.h

```diff
@@ -153,6 +154,35 @@ namespace mav {
             throw std::runtime_error("Unknown base type"); // should never happen
         }
 
+        uint64_t _computeSignatureHash48(const std::array<uint8_t, 32>& key) const {
+            // signature = sha256_48(secret_key + header + payload + CRC + link-ID + timestamp)
+            constexpr size_t maxSize = 32 + MessageDefinition::HEADER_SIZE +
+                                       MessageDefinition::MAX_PAYLOAD_SIZE +
+                                       MessageDefinition::CHECKSUM_SIZE + 1 + 6;
+            std::array<uint8_t, maxSize> data;
```

Comment:
Instead of copying the whole package here into an array to be hashed, you could just instantiate the hasher and feed it the respective parts one-by-one. Picosha seems to have an implementation for this.
Would be nicer implementation, but it seems that picosha is pretty inefficient anyways, so this probably doesn't make a real difference.

Reply:
Perhaps I'm not following correctly - but surely I need all the data in one place to be able to hash it here, right? I don't think I can feed it discontinuous blocks of memory.

Reply:
Ah yup, I figured it out.

Also I was having a bit of trouble finding an efficient way to feed the 6-byte timestamp into the hasher -- if there's a better way to do it than what I did, let me know.